configuration/src/main/java/io/camunda/configuration/beanoverrides/BrokerBasedPropertiesOverride.java

```diff
@@ -795,112 +785,146 @@ private void populateCamundaExporter(final BrokerBasedProperties override) {
-      setArg(args, "index.replicasByIndexName", database.getNumberOfReplicasPerIndex());
+      index.setReplicasByIndexName(database.getNumberOfReplicasPerIndex());
     }
     if (!database.getNumberOfShardsPerIndex().isEmpty()) {
-      setArg(args, "index.shardsByIndexName", database.getNumberOfShardsPerIndex());
+      index.setShardsByIndexName(database.getNumberOfShardsPerIndex());
     }
+  }
 
-    setArg(
-        args, "history.processInstanceEnabled", database.getHistory().isProcessInstanceEnabled());
-
-    setArg(args, "history.retention.policyName", database.getHistory().getPolicyName());
-    setArg(args, "history.elsRolloverDateFormat", database.getHistory().getElsRolloverDateFormat());
-    setArg(args, "history.rolloverInterval", database.getHistory().getRolloverInterval());
-    setArg(args, "history.rolloverBatchSize", database.getHistory().getRolloverBatchSize());
-    setArg(
-        args,
-        "history.waitPeriodBeforeArchiving",
-        database.getHistory().getWaitPeriodBeforeArchiving());
-    setArg(
-        args, "history.delayBetweenRuns", database.getHistory().getDelayBetweenRuns().toMillis());
-    setArg(
-        args,
-        "history.maxDelayBetweenRuns",
-        database.getHistory().getMaxDelayBetweenRuns().toMillis());
-
-    setArg(args, "createSchema", database.isCreateSchema());
+  private void applyCamundaExporterConfigurationHistory(
+      final HistoryConfiguration history, final DocumentBasedSecondaryStorageDatabase database) {
+    history.setProcessInstanceEnabled(database.getHistory().isProcessInstanceEnabled());
+    history.getRetention().setPolicyName(database.getHistory().getPolicyName());
+    history.setElsRolloverDateFormat(database.getHistory().getElsRolloverDateFormat());
+    history.setRolloverInterval(database.getHistory().getRolloverInterval());
+    history.setRolloverBatchSize(database.getHistory().getRolloverBatchSize());
+    history.setWaitPeriodBeforeArchiving(database.getHistory().getWaitPeriodBeforeArchiving());
+    history.setDelayBetweenRuns(
+        Math.toIntExact(database.getHistory().getDelayBetweenRuns().toMillis()));
+    history.setMaxDelayBetweenRuns(
+        Math.toIntExact(database.getHistory().getMaxDelayBetweenRuns().toMillis()));
+  }
 
-    if (database.getIncidentNotifier() != null) {
-      setArg(args, "notifier.webhook", database.getIncidentNotifier().getWebhook());
-      setArg(args, "notifier.auth0Domain", database.getIncidentNotifier().getAuth0Domain());
-      setArg(args, "notifier.auth0Protocol", database.getIncidentNotifier().getAuth0Protocol());
-      setArg(args, "notifier.m2mClientId", database.getIncidentNotifier().getM2mClientId());
-      setArg(args, "notifier.m2mClientSecret", database.getIncidentNotifier().getM2mClientSecret());
-      setArg(args, "notifier.m2mAudience", database.getIncidentNotifier().getM2mAudience());
-    }
+  private void applyCamundaExporterConfigurationIncidentNotifier(
```

Comment:
🔧 maybe instead of naming all these methods with `...CamundaExporter...`, we could extract them to their own class?